lib/Mojo/Promise.pm

```diff
@@ -41,6 +42,19 @@ sub finally {
   return $new;
 }
 
+sub get {
+  my ($self) = @_;
+  croak q{'get' cannot be called when the event loop is running}
+    if (my $loop = $self->ioloop)->is_running;
+  my ($res, $err);
+  $self->then(sub { $res = [@_]; $loop->stop },
+    sub { $err = [@_]; $loop->stop });
+  $loop->start;
```

Comment:
Opinion: should this be `$loop->start until $res or $err;` so that it is not interrupted by something else stopping the loop?